src/solvers/dgsem_tree/dg_2d_parabolic.jl

```diff
@@ -0,0 +1,652 @@
+    # into FMAs (see comment at the top of the file).
+    factor_1 = boundary_interpolation[1,          1]
+    factor_2 = boundary_interpolation[nnodes(dg), 2]
+    @threaded for element in eachelement(dg, cache)
+      for l in eachnode(dg)
+        for v in eachvariable(equations_parabolic)
+          let du = u_grad[1]
+            # surface at -x
+            du[v, 1,          l, element] = (
+              du[v, 1,          l, element] - surface_flux_values[v, l, 1, element] * factor_1)
+
+            # surface at +x
+            du[v, nnodes(dg), l, element] = (
+              du[v, nnodes(dg), l, element] + surface_flux_values[v, l, 2, element] * factor_2)
+          end
+
+          let du = u_grad[2]
+            # surface at -y
+            du[v, l, 1,          element] = (
+              du[v, l, 1,          element] - surface_flux_values[v, l, 3, element] * factor_1)
+
+            # surface at +y
+            du[v, l, nnodes(dg), element] = (
+              du[v, l, nnodes(dg), element] + surface_flux_values[v, l, 4, element] * factor_2)
+          end
+        end
+      end
+    end
+  end
+
+  # Apply Jacobian from mapping to reference element
+  @trixi_timeit timer() "Jacobian" begin
+    apply_jacobian!(u_grad[1], mesh, equations_parabolic, dg, cache_parabolic)
+    apply_jacobian!(u_grad[2], mesh, equations_parabolic, dg, cache_parabolic)
+  end
+
+  return nothing
+end
+
+
+# This method is called when a SemidiscretizationHyperbolic is constructed.
+# It constructs the basic `cache` used throughout the simulation to compute
+# the RHS etc.
+function create_cache_parabolic(mesh::TreeMesh{2}, equations_hyperbolic::AbstractEquations,
+                                equations_parabolic::AbstractEquationsParabolic,
+                                dg::DG, dg_parabolic, RealT, uEltype)
+  # Get cells for which an element needs to be created (i.e. all leaf cells)
+  leaf_cell_ids = local_leaf_cells(mesh.tree)
+
+  elements = init_elements(leaf_cell_ids, mesh, equations_hyperbolic, dg.basis, RealT, uEltype)
+
+  n_vars = nvariables(equations_hyperbolic)
+  n_nodes = nnodes(elements)
+  n_elements = nelements(elements)
+  u_transformed = Array{uEltype}(undef, n_vars, n_nodes, n_nodes, n_elements)
+  u_grad = ntuple(_ -> similar(u_transformed), ndims(mesh))
+  viscous_flux = ntuple(_ -> similar(u_transformed), ndims(mesh))
+
+  interfaces = init_interfaces(leaf_cell_ids, mesh, elements)
+
+  boundaries = init_boundaries(leaf_cell_ids, mesh, elements)
+
+  # mortars = init_mortars(leaf_cell_ids, mesh, elements, dg.mortar)
+
+  # cache = (; elements, interfaces, boundaries, mortars)
+  cache = (; elements, interfaces, boundaries, u_grad, viscous_flux, u_transformed)
+
+  # Add specialized parts of the cache required to compute the mortars etc.
+  # cache = (;cache..., create_cache(mesh, equations_parabolic, dg.mortar, uEltype)...)
+
+  return cache
+end
+
+
+# Needed to *not* flip the sign of the inverse Jacobian
```

Comment:
Could you please remind me again why we have this difference? I think besides the reason for adding the function, we should also document the reason for requiring the the sign not to be flipped.

Reply:
I think it's because of the convention that the Navier-Stokes system is in the form
```math
du/dt + df/dx = dg/dx + f(x,t)
```
where `g(u)` is the viscous flux we define. We could either add this to the comments or flip the sign convention (I prefer adding it to the comments).

Reply:
Haha, now I understand - thanks for the clarification 🙏 

However, for general systems, wouldn't 

$$
\frac{\partial u}{\partial t} + \frac{\partial}{\partial x} f(u)  + \frac{\partial}{\partial x}  g(u, \nabla u) = f(x,t)
$$

make more sense? Considering that there's a myriad of systems, in addition to this introducing a new potential source of errors, I'd like to keep the code as simple as possible.

Reply:
I usually prefer to write

$$
\partial_t u = \Delta u
$$

for the heat equation. From my point of view, it is fine to choose either convention - we just need to document it and stick to it. The current version has the nice benefit that it allows us to build matrix-free operators for elliptic problems without additional sign change.

Reply:
Added some documentation in https://github.com/trixi-framework/Trixi.jl/commit/299f04e4c91c73a5fe66e81007a256b7d1027cb4. 

Is this OK to resolve?

Reply:
👍 Thanks.